libsolidity/ast/Types.cpp

```diff
@@ -3571,7 +3574,7 @@ string FunctionType::externalSignature() const
 
 	// "inLibrary" is only relevant if this is not an event.
 	bool inLibrary = false;
-	if (kind() != Kind::Event)
+	if (kind() != Kind::Event && m_declaration->scope())
```

Comment:
Can you check if this is needed after https://github.com/ethereum/solidity/pull/9586 ?